src/types.ts

```diff
@@ -63,5 +63,6 @@ export type GetObjectId = (
   object: {
     readonly id?: string;
     readonly __typename?: string;
-  }
-) => string;
+  },
+  path: ReadonlyArray<string>
+) => string | undefined;
```

Comment:
@jonaskello maytbe it would be clearer if `getObjectToId` returned a "resultobject" which could look like

```ts
export type ObjectToIdResult = ObjectToIdResultSuccess | ObjectToIdResultFailure

export interface ObjectToIdResultSuccess {
    readonly resolved: true;
    readonly id: string
}

export interface ObjectToIdResultFailure {
    readonly resolved: false;
}

```

Reply:
Maybe we should have som performance tests what implications code changes has